cartridges/plugin_beam/cartridge/scripts/utils/getFinalLanguageCode.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * This helper function checks if the current language is supported by Beam.
+ * If the current language is supported, it returns it, otherwise, it returns the default language.
+ *
+ * @param {dw.system.Site} site The current site object
+ * @param {string} currentLanguageCode The current language code (e.g., 'en', 'fr', etc.)
+ * @returns {string} The final language code
+ */
+function getFinalLanguageCode(site, currentLanguageCode) {
+    var supportedLangString =
+        site.getCustomPreferenceValue("beamSupportedLang");
+    var supportedLang = [];
+    try {
+        supportedLang = JSON.parse(supportedLangString);
+    } catch (e) {
+        var Logger = require("dw/system/Logger");
+        Logger.error("Error parsing beamSupportedLang: {0}", e.message);
```

Comment:
`suggestion` I may leave this out of the Logger -- if we end up finding a language that doesn't parse correctly, it seems like it will fill up a partner's logs very quickly.